cmake/mpi.cmake

```diff
@@ -18,4 +18,16 @@ if(ENABLE_MPI)
     set(MPI_LANGUAGE C)
   endif()
 
+  if(MSVC)
+    list(APPEND TPL_DEFINES -D_SCL_SECURE_NO_WARNINGS)
+    list(APPEND TPL_DEFINES -D_CRT_SECURE_NO_WARNINGS)
+    list(APPEND TPL_DEFINES -D_SCL_SECURE_NO_DEPRECATE)
+    list(APPEND TPL_DEFINES -D_CRT_SECURE_NO_DEPRECATE)
+    list(APPEND TPL_DEFINES -D_CRT_NONSTDC_NO_WARNINGS)
+    list(APPEND TPL_DEFINES -D_HAS_AUTO_PTR_ETC=1)
+    list(APPEND TPL_DEFINES -D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING)
+    list(APPEND TPL_DEFINES -D_SILENCE_CXX17_ALLOCATOR_VOID_DEPRECATION_WARNING)
```

Comment:
Do we really produce _all_ of these kinds of warnings?

Reply:
I admit I just copied the defines from HPX. I will prune those down to a minimum.